python_modules/dagster/dagster/_core/definitions/metadata/__init__.py

```diff
@@ -811,8 +811,8 @@ def __new__(cls, asset_key: "AssetKey"):
 
     @public
     @property
-    def value(self) -> "AssetKey":
-        return self.value
+    def value(self) -> Self:
+        return self
```

Comment:
ah you're right. Interesting though - I saw this pattern in a good number of places of value returning self - should we change those callsites too, or are those special cases? Offenders are `TableMetadataValue` and `PythonArtifactMetadataValue`

Reply:
Yeah I can see those two cases-- it's not good, but I think the reason we do it there is because those classes actually contain multiple fields, so there is no other object we can return for `value` that contains all of the wrapped info.

Reply:
gotcha - makes sense